set.mm

```diff
@@ -50673,6 +50733,14 @@ Contrast with domain (defined in ~ df-dm ).  For alternate definitions,
     ( wss ccnv cdm crn cnvss dmss syl df-rn 3sstr4g ) ABCZADZEZBDZEZAFBFLMOCNPC
     ABGMOHIAJBJK $.
 
+  ${
+    rnssi.1 $e |- A C_ B $.
+    $( Subclass inference for range.  (Contributed by Peter Mazsa,
+       24-Sep-2022.) $)
+    rnssi $p |- ran A C_ ran B $=
+      ( wss crn rnss ax-mp ) ABDAEBEDCABFG $.
+  $}
```

Comment:
Used only once ? You should try to `scripts/min.cmd` to `minimize_with rnssi` theorems currently using `rnss`, and if (and only if) it decreases proofs by a total of more bytes than it takes to state and prove rnssi, then add it (and for ease of reviewing, in its own PR or at least its own commit).

Reply:
I like this policy: do not introduce an alternate form to the default (general) form of a theorem until you prove that its net effect to the size of the database is not positive. Based on this maybe we can even delete some inference forms and reduce the size of the database as well! I think @wlammen would like this (in the context of https://github.com/metamath/set.mm/pull/2867#pullrequestreview-1127557358 ), couldn't we make this an explicit convention?
I'll have time to experiment with this at the weekend.

Reply:
I don't know whether it is written in ~conventions (if not, I'll add it), but it has been a rule of thumb for quite some time, initiated by Norm, who used the phrase "[the lemma] pays for itself".  I would consider it as a sufficient condition for moving a lemma to Main (but not a necessary condition: a statement can be interesting in itself, without being widely used). Also, I would be more reluctant to remove a theorem from Main, even if I would not have introduced it in the first place (that is: "now that it's here, let it be").

Reply:
The wording in conventions is "Usually we will also automatically accept a _new_ theorem that is used to shorten multiple proofs, if the total size of set.mm (including the comment of the new theorem, not including the acknowledgment) decreases as a result. "

Reply:
Simple method, thank you the idea @avekens . I checked cca. 20% of possible cases and found useful in 5 of them: ~ ssrnres ~ fssres ~ smores ~ brdom4 ~ smobeth . So it seems that we need this inference. Based on that I mark this resolved.

Reply:
Let's say it's ok here.
If you want to do this properly in the future, you have to check that it shortens proofs by more bytes than the number of bytes (characters) necessary to state and prove the result, as I recalled above, and as was also recalled by @jkingdon (e.g., do as in https://github.com/metamath/set.mm/pull/2796#discussion_r955343777 where you see the output of `scripts/min.cmd`)